service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/AzureStorageAccessService.java

```diff
@@ -31,43 +24,38 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
+/**
+ * Service for Azure storage access management.
+ *
+ * <p>Rather than providing direct access to Azure storage containers, storage access is
+ * accomplished via the minting of shared access signatures (SAS).
+ */
 @Component
-public class AzureControlledStorageResourceService {
+public class AzureStorageAccessService {
 
   public record AzureSasBundle(String sasToken, String sasUrl) {}
 
   private final SamService samService;
-  private final AzureCloudContextService azureCloudContextService;
-  private final CrlService crlService;
-  private final ControlledResourceService controlledResourceService;
-  private final AzureConfiguration azureConfiguration;
   private final FeatureConfiguration features;
+  private final StorageAccountKeyProvider storageAccountKeyProvider;
 
   @Autowired
-  public AzureControlledStorageResourceService(
+  public AzureStorageAccessService(
       SamService samService,
-      AzureCloudContextService azureCloudContextService,
-      CrlService crlService,
-      ControlledResourceService controlledResourceService,
-      AzureConfiguration azureConfiguration,
+      StorageAccountKeyProvider storageAccountKeyProvider,
       FeatureConfiguration features) {
     this.samService = samService;
-    this.azureCloudContextService = azureCloudContextService;
```

Comment:
These dependencies are no longer needed now that the key retrieval has been factored into a separate class.